### DISCUSSION TITLE
Would a Raspberry Pi 3 or 4 (or other similar SBC) run this OK?

### DISCUSSION THREAD

**Discussion by diveleader** [jasonacox/Powerwall-Dashboard#282] [Q&A] [answered]
I've got my dashboard up and running on an old repurposed chromebook.
I've enjoyed the tinkering to get it to work and am thinking, of perhaps repeating it with a single board computer.

Looking at Aliexpress, Orange Pi 3 LTS (2 gig ram / All winner H6) looks a good price, but would it run OK?
I'm not looking to do anything else with the SBC, just monitor my Powerwall.

The next step up would be to a Orange Pi 4 (4 gig ram / RK 3399) but, % wise, its a step up in cost.

Any thoughts or suggestions for a low priced board that can run the Dashboard?

**Comment by jasonacox** [comment]
The Orange Pi 3 LTS has a Cortex-A53 CPU which appears to be compatible with arm64 binaries. The docker containers we use for this project are set up to include both armv7 (32-bit) and arm64 containers. So that part looks positive. The OS options for this look to be only from Orange (?) but does include a Ubuntu image which should work well.  If you do get one and get it to work, let us know!

Anyone else have a working Orange Pi setup?

**Comment by anderep** [comment]
Why not raspberry pi?  I trust the Brit’s more then china…

**Reply by jasonacox** [reply]
My test rig is a Raspberry Pi 4.  Works great!  Highly recommended.

**Reply by jasonacox** [reply]
Hijacking this discussion a bit... my test rigs are set to validate changes on these platforms, so these are our "assert to work" platforms:

- Standard Docker (not rootless)
- Raspberry Pi
- Ubuntu Linux
- Windows 10/11 with WSL (complicated because the MS Linux support requires additional setup)
- MacOS
- No installations where there are other InfluxDB, Grafana, Telegraf instances on the same host

From successes reported by the community, we could list, "can work with user customization":

- Rootless Docker
- Synology NAS
- Linux (outside of Ubuntu LTS) or other *nix
- Shared InfluxDB 1.8/2.x, Grafana, Telegraf instances on same host

What am I missing or what should we adjust?

**Reply by rvry** [reply]
> My test rig is a Raspberry Pi 4. Works great! Highly recommended.

Would you recommend 2, 4 or 8 Gb Raspberry Pi 4?

I've started using this dashboard (it's fantastic!) but running on my MacOSX laptop. It stops collected data when laptop goes to sleep so I want something running permanently :-)

**Reply by jasonacox** [reply]
Ha!  I hear you!  I have both a 2G and 4G version, both work great.  The saying is true, the more memory the better. I always try to go for the highest I can afford... I've never regretted that. 😁

**Comment by diveleader** [comment, marked as answer]
An update,
I purchased a Orange Pi4 LTS from Aliexpress. It has a Rockchip 6 core RK3399 SOC with 4 gig of RAM and 16 gig EMMC.

I installed Armbian (it was by  far the fastest of the distros I installed).

Installation of the Dashboard went smoothly. 

It runs the dashboard perfectly fine and draws 1.5-2.5 watts in operation

**Reply by jasonacox** [reply]
Thanks, @diveleader !  That's great to know.
